File: 백준/Silver/1181. 단어 정렬/단어 정렬.py
import sys

N = int(sys.stdin.readline())
obj = {}
for i in range(N) :
  string = input()
  obj[string] = len(string)

key_sort = dict(sorted(obj.items()))
value_sort = sorted(key_sort, key=lambda x : key_sort[x])
print("\n".join(value_sort))

File: 백준/Silver/1181. 단어 정렬/단어 정렬.js
const readLine = require('fs').readFileSync('dev/stdin').toString().trim();
const [N, ...input] = readLine.split('\n');

const noDuplicate = [...new Set(input)];

const sortArr = noDuplicate.sort((a, b) => {
  if (a.length > b.length) return 1;
  else if (a.length < b.length) return -1;
  else return a.localeCompare(b);
});
console.log(sortArr.join('\n'));
